### ISSUE TITLE
CORS configuration for S3 bucket is not working

### ISSUE THREAD

**Issue by hminh21** [localstack/localstack#2168]
For personal reasons, I pull the latest codebase from `master` to infra S3 service by myself. Everything works well (get/put object) except the cors configuration. I just wanna allow some origins to be able to upload to my bucket but it doesn't work although the cors configuration was set. Here are my cors.json file to set cors:
```
{
    "CORSRules":[
       {
          "AllowedOrigins":["https://exampleA.com", "https://exampleB.com"],
          "AllowedHeaders":[
             "x-amz-tagging"
          ],
          "AllowedMethods":["GET", "PUT"],
          "MaxAgeSeconds":3000,
          "ExposeHeaders": ["x-amz-tagging"]
       } 
    ]
 }

```
I use this command and it updated successfully:
`aws --endpoint-url=http://localhost:4572 s3api put-bucket-cors --bucket $AWS_BUCKET --cors-configuration file://cors.json`
I have double-checked by using get-bucket-cors command and it shows the correct configuration
`aws --endpoint-url=http://localhost:4572 s3api get-bucket-cors --bucket $AWS_BUCKET`

When I tried to upload file to S3 bucket from `https://exampleC.com` which were not listed in the allowed configuration:
- **Expected result**: Uploading will be failed due to being blocked by CORS
- **Actual result**: Uploading is still successful and the response returns `Access-Control-Allow-Origin: https://exampleC.com`

Have I missed something or S3 localstack does not support cors configuration? Please let me know asap, thank you.

**Comment by leikoilja** [comment]
@usmangani1, I am afraid the problem still persist.

I am on the latest `localstack` docker image and when i have a CORS configuration like this:
```json
{
  "CORSRules": [
    {
      "AllowedOrigins": ["http://www.example.com"],
      "AllowedHeaders": ["Authorization"],
      "AllowedMethods": ["GET", "HEAD"],
      "MaxAgeSeconds": 3000
    },
    {
      "AllowedOrigins": ["http://www.example.com"],
      "AllowedHeaders": ["Authorization"],
      "AllowedMethods": ["POST"],
      "MaxAgeSeconds": 3000
    }
  ]
}
```
And when i do `POST` request from my `localhost` it still allows it. I would expect it to fail due to CORS, since my origin is not `example.com`.

**Comment by whummer** [comment]
Thanks for the update @leikoilja @hminh21 . This should now (hopefully) be fixed in #3204 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks!
